src/sql/workbench/services/notebook/browser/notebookServiceImpl.ts

```diff
@@ -407,8 +407,7 @@ export class NotebookService extends Disposable implements INotebookService {
 		if (!notebookUri) {
 			return undefined;
 		}
-		// The NotebookEditor will not be found if there is query or fragments attached to the URI
-		let uriString = notebookUri.with({ query: '', fragment: '' }).toString();
+		let uriString = notebookUri.toString();
```

Comment:
I'm not sure I understand what the issue was here. Why wasn't the old logic working?

Reply:
I stated the issue here:
After the parameterized notebook was opened, I clicked run all cells, it said "unable to runAllCells of undefined". 
As such, I debugged and found runAllCells function call findNotebook, which then alters the URI as done previously by removing the query and fragment, but then it is compared to the id retrieved by the call here -
let editor = this.listNotebookEditors().find(n => n.id === uriString);.
The id contains the URI and query, as such it would return an undefined editor.
In order to fix that I would have to edit the listNotebookEditors function as well to edit the id (which is the same format of uriString - which to me didn't seem reasonable).

Reply:
The ID of the editor contains the query? You were stripping out the query/fragment of the passed in URI before so I assume that's the case...

If so that sounds like the issue I was concerned about with originally - that we're using these URIs in different ways in different places which is going to keep being an annoyance (and likely causing bugs).

Reply:
Yes the id of the editor contains the query and so it would not find the editor in the listNotebookEditors when comparing n.id === uriString. For now, I reverted back to not changing these functions (in order to not cause other bugs) but can come back at a later date and see what to change.

Reply:
Alright, we should definitely discuss what should happen here. Because typically I expect the editors to fall into two buckets : 

1. Parameterized editors (have parameters passed in) - these I would expect to always open as untitled documents, and if saved not persist the parameters in the URI (they would be inserted into the Notebook itself though if the user wanted to save it then)
2. Editors for Notebooks on disk - I would never expect these to have parameters passed in via the URI since that should always open an Untitled editor

If that doesn't match what you were thinking we should definitely chat.

Reply:
I agree with those two buckets of editors, would you say that I would go and edit the logic in listNotebooks then?

Reply:
I don't think listNotebookEditors is the problem - that's just using editor.getID. I would instead be looking at making sure the editors are having their ID set correctly such that the query isn't included in that.